src/Octoshift/Services/GithubApi.cs

```diff
@@ -516,7 +517,11 @@ public virtual async Task<string> GetMigrationLogUrl(string org, string repo)
 
                 var nodes = (JArray)data["data"]["organization"]["repositoryMigrations"]["nodes"];
 
-                return nodes.Count == 0 ? null : (string)nodes[0]["migrationLogUrl"];
+                return nodes.Count == 0
+                    // No matching migration was found
+                    ? ((string MigrationLogUrl, string MigrationId)?)null
+                    // A matching migration was found, which may or may not have a migration log URL. If there is no migration log, it's an empty string.
+                    : (MigrationLogUrl: (string)nodes[0]["migrationLogUrl"], MigrationId: (string)nodes[0]["id"]);
```

Comment:
When we change the return type to a not nullable value tuple this should change too:
```suggestion
                return nodes.Count == 0
                    ? default
                    : (MigrationLogUrl: (string)nodes[0]["migrationLogUrl"], MigrationId: (string)nodes[0]["id"]);
```

also the code is clear enough here so no need to add comments. Generally speaking code comments are a smell to me which means the code is not clear enough so it needs to be re-written with better variable/method naming or more clear logic.

Reply:
Resolving as we agreed not to make the tuple non-nullable